samples/boards/nrf/nrfx_prs/src/main.c

```diff
@@ -221,7 +226,7 @@ static bool switch_to_uarte(void)
 	nrfx_uarte_config_t uarte_config = NRFX_UARTE_DEFAULT_CONFIG(
 		NRF_UARTE_PSEL_DISCONNECTED,
 		NRF_UARTE_PSEL_DISCONNECTED);
-	uarte_config.baudrate = NRF_UARTE_BAUDRATE_1000000;
+	uarte_config.baudrate = MHZ(1);
```

Comment:
You're right. thanks.

I tested with your workarounds an it works.

However to me it seems like this should be part of nrfx uninit function.
Clearing the event registers seems like an important step. For example if these functions are used in this scenario, or between images in a bootchain, like mcuboot -> tf-m -> application.

It can be added here as a temporary workaround until a new release of NRFX of course.